src/App.tsx

```diff
@@ -1,26 +1,361 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo, Filter, Erors } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components';
 
 export const App: React.FC = () => {
+  const input = useRef<HTMLInputElement>(null);
+  const hideErrorTimer = useRef<number | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoId, setDeletingTodoId] = useState<number[] | null>(null);
+  const [filter, setFilter] = useState(Filter.All);
+  const [title, setTitle] = useState('');
+  const [loadingError, setLoadingError] = useState({
```

Comment:
It's easier to save the correct error (enum value) or empty string if there is no error (empty string is a falsy value)